### PR TITLE
[VMRay] Add vmray-get-screenshots command

### PR DESCRIPTION
**Pull request by tkorten-vmray**
Add the vmray-get-screenshots command to retrieve all screenshots for one analysis.

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
Add the command `vmray-get-screenshots` to retrieve the screenshots which were made during a dynamic analysis. The screenshots are then returned to the war room.

Besides the new functionality the documentation was updated to contain the latest information.

## Screenshots
Screenshots sown in the war room:
Screenshots in war room

Screenshots and "Get Screenshots" command shown in the work plan:
grafik

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/VMRay/Integrations/VMRay/VMRay.py [hunk 1]
```diff
@@ -1342,6 +1363,38 @@ def get_summary_command():
...
+    analysis_id = demisto.args().get('analysis_id')
+    check_id(analysis_id)
+
+    screenshots_data = get_screenshots(analysis_id)
+
+    file_results = []
+    screenshot_counter = 0
+    try:
+        with ZipFile(io.BytesIO(screenshots_data), 'r') as screenshots_zip:
+            index_log_data = screenshots_zip.read('screenshots/index.log', pwd=DEFAULT_ARCHIVE_PASSWORD)
```
Comment (mmhw) [inline Packs/VMRay/Integrations/VMRay/VMRay.py:1376]: Will the password always be the default?
Can't the user change it, and then have to add a password parameter?

File: Packs/VMRay/ReleaseNotes/1_1_11.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+- Added the ***vmray-get-screenshots*** command to download the screenshots for a specific analysis.
+
+#### Playbooks
+
+##### Detonate File - VMRay
+
+Fixed an issue in which the playbook input parameters *archive_password*, *document_password*, *max_jobs*, *sample_type*, *shareable*, and *tags* were not used by the ***vmray-upload-sample*** command.
+
+<!--
```
Comment (mmhw) [inline Packs/VMRay/ReleaseNotes/1_1_11.md:15]: If all the changes in Playbook are just `format`, it seems better to revert it and cancel the changes

**Review by mmhw — COMMENTED** [review]
Hi @tkorten-vmray,
Thank you for your contribution!
Good work :)

Please note my small comments.

Please feel free to reach out to me with any questions - I'm available here or on slack :)
Thanks again

**Comment by mmhw** [comment]
> [CLA assistant check](https://cla-assistant.io/demisto/content?pullRequest=28871) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=28871) before we can accept your contribution.You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=28871) it.

Please fill the [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=28871)

**Comment by mmhw** [comment]
Hi @tkorten-vmray,
it looks good!

Have you checked after the last changes that nothing is broken and everything works as expected?
Also, please merge again from master.

Thanks!

**Comment by melamedbn** [comment]
Dear @tkorten-vmray,

I hope this message finds you well. Firstly, I want to express my appreciation for your efforts in contributing to the content. Your dedication to the project is evident, and I wanted to share a couple of suggestions to further enhance the quality of the work:

- Context Path Key Usage:
  - I noticed that you’ve put the inputs as plain text, I think using the complex context method could be better and minimize the risk of typos. To illustrate this, I've attached a screenshot for your reference.(Figure 1)
    - Playbook inputs: 'File' - This wasn't part of your initial contribution, but it can be helpful.
    - Task 6: I recommend revisiting the new inputs using this method.
  
image

Figure 1: Complex context vs plain text

- Typographical Correction:
  - In Task 14, I spotted a minor typo. It reads "Thread" when I believe it should be "Threat". Even though it wasn't part of the original contribution, addressing this can ensure clarity and precision.
    - If you decide to fix that, please note that an updated screenshot of the playbook will be needed as well.

I trust that these suggestions will be valuable in fine-tuning the content. Once again, thank you for your commitment and hard work. I'm confident that with these minor tweaks, we'll be able to deliver an even more polished product to our users.

Warm regards,
Ben

**Comment by kgal-pan** [comment]
@tkorten-vmray - Can you please sign the [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=28871).

**Comment by tkorten-vmray** [comment]
Hi @mmhw,
yes. both of the playbooks and the new command are tested and they are working as expected.
Master is now also merged into this branch.
Best regards

**Comment by melamedbn** [comment]
> Dear @melamedbn, thank you for the suggestions. I updated the `Detonate URL - VMRay` and `Detonate File - VMRay` playbooks so the typo is fixed in both and the complex input method is also used in both playbooks. The screenshots of both playbooks are also updated. Do I need to make another commit to update the path to the screenshots in the playbook README or is this done automatically? Regarding the license agreement, I'm currently requesting the approval internally. Best regards

@tkorten-vmray  Thank you for promptly addressing the feedback and making the necessary changes! And yes, the README should be updated manually as well using the generate-docs command.

**Comment by tkorten-vmray** [comment]
@melamedbn Thank you. I will update them. Are the images inside `docs/images/playbooks` then still required or is this an deprecated directory, which was replaced by `Packs/<name>/doc_files` and I can safely remove them?

**Review by melamedbn — APPROVED** [review]

**Comment by mmhw** [comment]
Hi @tkorten-vmray,
Good work, the PR is ready to merge.
It remains to fill the [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=28871).

Thanks

**Comment by mmhw** [comment]
Hi @tkorten-vmray,
The images of the playbooks should be inside the pack, in the `Packs/VMRay/doc_files` folder.
Thanks!

**Pull request merged**